packages/markdown-docx/src/ToOOXMLVisitor/index.js

```diff
@@ -86,21 +94,40 @@ class ToOOXMLVisitor {
                     let textValueTag = TEXT_RULE(subNode.text);
 
                     let tag = TEXT_WRAPPER_RULE(propertyTag, textValueTag);
+
+                    if (isLinkPropertyPresent) {
+                        // some rels are already present
+                        // To avoid overwrite ids start our
+                        // rels from 10
+                        let relationShipId = 'rId' + (this.relationships.length + 10).toString();
+                        tag = LINK_RULE(tag, relationShipId);
+                    }
                     this.tags = [...this.tags, tag];
                 } else if (this.getClass(subNode) === TRANSFORMED_NODES.code) {
                     let propertyTag = CODE_PROPERTIES_RULE();
+                    let isLinkPropertyPresent = false;
                     for (let property of properties) {
                         if (property === TRANSFORMED_NODES.emphasize) {
                             propertyTag += EMPHASIS_RULE();
                         } else if (property === TRANSFORMED_NODES.strong) {
                             propertyTag += STRONG_RULE();
+                        } else if (property === TRANSFORMED_NODES.link) {
+                            isLinkPropertyPresent = true;
+                            propertyTag += LINK_PROPERTY_RULE();
                         }
                     }
                     propertyTag = TEXT_STYLES_RULE(propertyTag);
 
                     let textValueTag = TEXT_RULE(subNode.text);
 
                     let tag = TEXT_WRAPPER_RULE(propertyTag, textValueTag);
+                    if (isLinkPropertyPresent) {
+                        // some inbuilt rels are already present for settings, theme, etc.
+                        // To avoid overwrite ids or conflicts start our
+                        // rels from 11
+                        let relationShipId = 'rId' + (this.relationships.length + 10).toString();
+                        tag = LINK_RULE(tag, relationShipId);
+                    }
```

Comment:
Why is there a repetition of the same code? Please follow DRY as much as possible.

Reply:
@algomaster99 
I decided to refactor this in a new PR. But will do in the same for now as it is related to `link` only.